Zattoo/Fastfile

```diff
@@ -0,0 +1,177 @@
+fastlane_version "2.66.2"
+
+require './helpers/zattoo.rb'
+
+#### Setup ####
+
+desc "Fetches development certificates and provisioning profiles"
+lane :fetch_dev_certificates do
+  match(type: 'development', readonly: true, team_id: Zattoo::TeamId, app_identifier: Zattoo::AppIdentifier::Production)
+  match(type: 'development', readonly: true, team_id: Zattoo::TeamId, app_identifier: Zattoo::TopShelfAppIdentifier::Production)
+end
+
+desc "Fetches distribution certificates and provisioning profiles"
+lane :fetch_distribution_certificates do
+  match(type: 'appstore', readonly: true, team_id: Zattoo::TeamId, app_identifier: Zattoo::AppIdentifier::Preview)
+  match(type: 'appstore', readonly: true, team_id: Zattoo::TeamId, app_identifier: Zattoo::AppIdentifier::AppleStore)
+  match(type: 'appstore', readonly: true, team_id: Zattoo::TeamId, app_identifier: Zattoo::AppIdentifier::Production)
+  match(type: 'appstore', readonly: true, team_id: Zattoo::TeamId, app_identifier: Zattoo::TopShelfAppIdentifier::Preview)
+  match(type: 'appstore', readonly: true, team_id: Zattoo::TeamId, app_identifier: Zattoo::TopShelfAppIdentifier::AppleStore)
+  match(type: 'appstore', readonly: true, team_id: Zattoo::TeamId, app_identifier: Zattoo::TopShelfAppIdentifier::Production)
```

Comment:
I actually like how the repetition makes it very explicit what happens. I would not do a loop here.

Reply:
I thought about that, but all the `match` parameters are the same except for the app identifier. Imo it doesn't really add any readability to have the explicit calls to match and I don't see any other advantages. 

The disadvantages I can see are that it's easy to forget a single app identifier (say, they add a watch extension) in that long list, and when a parameter gets added/changed it's easy to forget one in the middle of the 7 calls. 

Even if the company behind this has procedures in place to make sure these things never happen, since this is an example that should show other people getting started "how it's done", I would not generally recommend this style. 

I also just remembered that the loop isn't even necessary since `app_identifier` actually supports arrays as parameters 😄

Reply:
I personally still would leave it ;)

Reply:
I personally would still leave it the way it is.

Reply:
I personally would still leave it the way it is.

No matter if it is a loop (which many people working with Fastfile probably don't know as they have no idea of ruby) or "only" an array, both is not as explicit as seeing 4 lines of the same call, just with changing app identifiers.

Reply:
I also prefer them explicitly listed, it makes it more obvious, but I do also see the point on avoiding the repetition on the match parameters... 

So another option would be to keep the lane explicitly using each app identifier but creating a private lane to avoid the repetition on the parameters when using match:

```
lane :fetch_dev_certificates do
  run_match(type: 'development', app_identifier: Zattoo:AppIdentifier::Production)
  run_match(type: 'development', app_identifier: Zattoo:TopShelfAppIdentifier::Production)
end

desc "Fetches distribution certificates and provisioning profiles"
lane :fetch_distribution_certificates do
  run_match(type: 'appstore', app_identifier: Zattoo:AppIdentifier::Preview)
  run_match(type: 'appstore', app_identifier: Zattoo:AppIdentifier::AppleStore)
  run_match(type: 'appstore', app_identifier: Zattoo:AppIdentifier::Production)
  run_match(type: 'appstore', app_identifier: Zattoo:TopShelfAppIdentifier::Preview)
  run_match(type: 'appstore', app_identifier: Zattoo:TopShelfAppIdentifier::AppleStore)
  run_match(type: 'appstore', app_identifier: Zattoo:TopShelfAppIdentifier::Production)
end

private_lane :run_match do |options|
    type = options[:type]
    app_identifier = options[:app_identifier]
    match(type: type, readonly: true, team_id: Zattoo::TeamId, app_identifier: app_identifier)
end
```

What do you think?

Reply:
> I also prefer them explicitly listed, it makes it more obvious

Fair enough. I don't think it's necessary in this case since it doesn't really "fix" the repetition. Since you showed how to avoid repetition in this commit https://github.com/fastlane/examples/pull/38/commits/389bb2c7ef386eed260c7f17eed3a5bd1336f658 maybe just place a comment before the match calls that explains why? Something like 

```
# All app identifiers are passed to separate match calls to make it easier to see what is happening
```